src/Raven.Server/Documents/CompactDatabaseTask.cs

```diff
@@ -87,28 +104,39 @@ public CompactDatabaseTask(ServerStore serverStore, string database, Cancellatio
                     result.TreeName = null;
                     
                     _token.ThrowIfCancellationRequested();
-                    IOExtensions.MoveDirectory(basePath, basePath + "-old");
-                    IOExtensions.MoveDirectory(basePath + "-Compacting", basePath);
-
-                    var oldIndexesPath = new PathSetting(basePath + "-old").Combine("Indexes");
-                    var newIndexesPath = new PathSetting(basePath).Combine("Indexes");
-                    IOExtensions.MoveDirectory(oldIndexesPath.FullPath, newIndexesPath.FullPath);
 
-                    var oldConfigPath = new PathSetting(basePath + "-old").Combine("Configuration");
-                    var newConfigPath = new PathSetting(basePath).Combine("Configuration");
-                    IOExtensions.MoveDirectory(oldConfigPath.FullPath, newConfigPath.FullPath);
-                }
-                catch (Exception e)
-                {
-                    throw new InvalidOperationException($"Failed to execute compaction for {_database}", e);
-                }
-                finally
+                    SwitchDatabaseDirectories(basePath, tmpDirectory, compactDirectory);
```

Comment:
I wonder if we should call `EnsureReadWritePermissionForDirectory` again here before `SwitchDatabaseDirectories`? The compaction can take hours for huge dbs, so just to ensure nothing has changed meanwhile and we still have an access

Reply:
Might be a good idea, but if they changed the permissions midway, that is PEBKAC